src/shared/TermsAndConditionModal/TermsAndConditionModal.tsx

```diff
@@ -8,21 +8,37 @@ import {
 } from "@patternfly/react-core";
 import { useTranslation } from "react-i18next";
 
+type OuiaIdModal = {
+  ouiaIdModal?: string;
+  ouiaIdButtonViewTerms?: string;
+  ouiaIdButtonCancel?: string;
+};
+
 export type TermsAndConditionModalProps = {
   onClickViewTermsConditions: () => void;
   onCancel: () => void;
   isModalOpen: boolean;
   serviceName: string;
+  ouiaIds?: OuiaIdModal;
```

Comment:
It's better to avoid passing objects as props. If the object isn't static, it will make the component re-render every time the parent component does.